src/table.js

```diff
@@ -66,7 +66,12 @@ function objectHasEnumerableKeys(value) {
 }
 
 function isQueryResultSetSchema(schemas) {
-  return (Array.isArray(schemas) && schemas.every((s) => s && typeof s.name === "string"));
+  return (
+    Array.isArray(schemas) &&
+    schemas.every(
+      (s) => s && typeof s.name === "string" && typeof s.type === "string"
```

Comment:
Totally optional, but for readability, we could break this out into a separate function:

```js
function isColumnSchema(schema) {
  return schema && typeof schema.name === "string" && typeof schema.type === "string"
}
```